src/CMakeLists.txt

```diff
@@ -0,0 +1,272 @@
+        widgets/dialogs/IrcConnectionEditor.{hpp,cpp}
+        widgets/dialogs/LastRunCrashDialog.{hpp,cpp}
+        widgets/dialogs/LoginDialog.{hpp,cpp}
+        widgets/dialogs/NotificationPopup.{hpp,cpp}
+        widgets/dialogs/QualityPopup.{hpp,cpp}
+        widgets/dialogs/SelectChannelDialog.{hpp,cpp}
+        widgets/dialogs/SelectChannelFiltersDialog.{hpp,cpp}
+        widgets/dialogs/SettingsDialog.{hpp,cpp}
+        widgets/dialogs/TextInputDialog.{hpp,cpp}
+        widgets/dialogs/UpdateDialog.{hpp,cpp}
+        widgets/dialogs/UserInfoPopup.{hpp,cpp}
+        widgets/dialogs/WelcomeDialog.{hpp,cpp}
+        widgets/dialogs/switcher/NewTabItem.{hpp,cpp}
+        widgets/dialogs/switcher/QuickSwitcherModel.{hpp,cpp}
+        widgets/dialogs/switcher/QuickSwitcherPopup.{hpp,cpp}
+        widgets/dialogs/switcher/SwitchSplitItem.{hpp,cpp}
+        widgets/helper/Button.{hpp,cpp}
+        widgets/helper/ChannelView.{hpp,cpp}
+        widgets/helper/ColorButton.{hpp,cpp}
+        widgets/helper/ComboBoxItemDelegate.{hpp,cpp}
+        widgets/helper/DebugPopup.{hpp,cpp}
+        widgets/helper/EditableModelView.{hpp,cpp}
+        widgets/helper/EffectLabel.{hpp,cpp}
+        widgets/helper/NotebookButton.{hpp,cpp}
+        widgets/helper/NotebookTab.{hpp,cpp}
+        widgets/helper/QColorPicker.{hpp,cpp}
+        widgets/helper/ResizingTextEdit.{hpp,cpp}
+        widgets/helper/ScrollbarHighlight.{hpp,cpp}
+        widgets/helper/SearchPopup.{hpp,cpp}
+        widgets/helper/SettingsDialogTab.{hpp,cpp}
+        widgets/helper/SignalLabel.{hpp,cpp}
+        widgets/helper/TitlebarButton.{hpp,cpp}
+        widgets/listview/GenericItemDelegate.{hpp,cpp}
+        widgets/listview/GenericListItem.{hpp,cpp}
+        widgets/listview/GenericListModel.{hpp,cpp}
+        widgets/listview/GenericListView.{hpp,cpp}
+        widgets/settingspages/AboutPage.{hpp,cpp}
+        widgets/settingspages/AccountsPage.{hpp,cpp}
+        widgets/settingspages/CommandPage.{hpp,cpp}
+        widgets/settingspages/ExternalToolsPage.{hpp,cpp}
+        widgets/settingspages/FiltersPage.{hpp,cpp}
+        widgets/settingspages/GeneralPage.{hpp,cpp}
+        widgets/settingspages/GeneralPageView.{hpp,cpp}
+        widgets/settingspages/HighlightingPage.{hpp,cpp}
+        widgets/settingspages/IgnoresPage.{hpp,cpp}
+        widgets/settingspages/KeyboardSettingsPage.{hpp,cpp}
+        widgets/settingspages/ModerationPage.{hpp,cpp}
+        widgets/settingspages/NotificationPage.{hpp,cpp}
+        widgets/settingspages/SettingsPage.{hpp,cpp}
+        widgets/splits/ClosedSplits.{hpp,cpp}
+        widgets/splits/EmoteInputItem.{hpp,cpp}
+        widgets/splits/EmoteInputPopup.{hpp,cpp}
+        widgets/splits/Split.{hpp,cpp}
+        widgets/splits/SplitContainer.{hpp,cpp}
+        widgets/splits/SplitHeader.{hpp,cpp}
+        widgets/splits/SplitInput.{hpp,cpp}
+        widgets/splits/SplitOverlay.{hpp,cpp}
+        widgets/dialogs/IrcConnectionEditor.ui
+        )
```

Comment:
Only .cpp files should be part of the `add_executable` code (and .ui/qrc etc)

Reply:
The header files should be added to `add_executable` to appear as dependencies in the Makefile, and also to be added in the Visual Studio project (otherwise the header files will be marked as external dependencies)

Reply:
SeemsGood :+1: been taught wrong, googling modern examples shows you're right